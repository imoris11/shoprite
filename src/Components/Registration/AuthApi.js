export const loginApi = async (url, data) => {
  let response = await fetch(url, {
    method:'POST',
    body: JSON.stringify(data),
    headers:{'content-type': 'application/json'}
  })
  let result =  await response.json();
  if (result.error) {
      return result;
    }else{
      localStorage.setItem('token', result.accessToken);
      localStorage.setItem('customer_id', result.user.customer_id)
      result.user.firstName = result.user.name.split(" ")[0];
      result.user.lastName = result.user.name.split(" ")[1];
      localStorage.setItem('customer', JSON.stringify(result.user));
      window.location.reload();
    }
}

export const registerApi = async (url, data) => {
  let response = await fetch(url, {
    method:'POST',
    body: JSON.stringify(data),
    headers:{'content-type': 'application/json'}
    })
  let result = await response.json();
  if (result.error){
    return result;
  }else{
    localStorage.setItem('token', result.accessToken);
    localStorage.setItem('customer_id', result.customer.customer_id)
    result.customer.firstName = result.customer.name.split(" ")[0];
    result.customer.lastName = result.customer.name.split(" ")[1];
    localStorage.setItem('customer', JSON.stringify(result.customer));
    window.location.reload();
  }
}
