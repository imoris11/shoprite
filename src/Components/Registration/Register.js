import React, {Component} from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import {Row, Col} from 'react-bootstrap';
class Register extends Component {
  state = {
    loading:false
  }
  handleSubmit = (e) => {
    e.preventDefault();
    this.setState(prevState => ({loading:!prevState.loading, form_submitted:true}));
    let data = {
      name:this.state.name,
      email:this.state.email,
      password:this.state.password
    };
    let url = 'https://backendapi.turing.com/customers/';
    fetch(url, {
      method:'POST',
      body: JSON.stringify(data),
      headers:{'content-type': 'application/json'}
    })
    .then(response => response.json())
    .then(result => {
      if (result.error) {
        this.setState({errorMessage:result.error.message});
      }else{
        localStorage.setItem('token', result.accessToken);
        localStorage.setItem('customer_id', result.customer.customer_id)
        result.customer.firstName = result.customer.name.split(" ")[0];
        result.customer.lastName = result.customer.name.split(" ")[1];
        localStorage.setItem('customer', JSON.stringify(result.customer));
        console.log(result);
        window.location.reload();
      }
      this.setState(prevState => ({loading:!prevState.loading}));
    })
    .catch(error => {
      this.setState({errorMessage:"Error registering, please try again.", loading:false});
    });

  }
  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }
  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <div style={{marginTop:-40}}>
          <div>
            <TextField
              id="standard-name"
              label="Name"
              name="name"
              margin="normal"
              style={{width:320}}
              onChange={this.handleChange}
            />
          </div>
          <div>
            <TextField
              id="standard-email"
              label="Email"
              name="email"
              margin="normal"
              onChange={this.handleChange}
              style={{width:320}}
            />
          </div>
          <div>
            <TextField
              id="standard-password-input"
              name="password"
              label="Password"
              type="password"
              autoComplete="current-password"
              margin="normal"
              onChange={this.handleChange}
              style={{width:320}}
            />
          </div>
          <Row style={{marginTop:20}}>
            <Col>
                <span className='text-danger'>{this.state.errorMessage}</span>
              {this.state.loading ? <Button className='float-right' variant="contained" color="secondary" >
                Signing up...
              </Button> :
              <Button id="submit_form" className='float-right' type='submit' variant="contained" color="secondary" >
                Sign Up
              </Button>}

            </Col>
          </Row>
        </div>
      </form>
    )
  }
}

export default Register;
