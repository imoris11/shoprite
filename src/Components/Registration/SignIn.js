import React, {Component} from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import {Row, Col} from 'react-bootstrap';
class SignIn extends Component {
  state = {
    loading:false
  }
  handleSubmit = (e) => {
    e.preventDefault();
    this.setState(prevState => ({loading:!prevState.loading}));
    let data = {
      email:this.state.email,
      password:this.state.password
    };
    let url = 'https://backendapi.turing.com/customers/login';
    fetch(url, {
      method:'POST',
      body: JSON.stringify(data),
      headers:{'content-type': 'application/json'}
    })
    .then(response => response.json())
    .then(result => {
      if (result.error) {
        this.setState({errorMessage:result.error.message});
      }else{
        localStorage.setItem('token', result.accessToken);
        localStorage.setItem('customer_id', result.user.customer_id)
        result.user.firstName = result.user.name.split(" ")[0];
        result.user.lastName = result.user.name.split(" ")[1];
        localStorage.setItem('customer', JSON.stringify(result.user));
        window.location.reload();
      }
      this.setState(prevState => ({loading:!prevState.loading}));
    })
    .catch(error => {
      this.setState({errorMessage:"Error logging in, please try again.", loading:false});
    });

  }
  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }
  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <div style={{marginTop:-40}}>
          <div>
            <TextField
              id="standard-uncontrolled"
              label="Email"
              margin="normal"
              name='email'
              style={{width:320}}
              onChange={this.handleChange}
            />
          </div>
          <div>
            <TextField
              id="standard-password-input"
              label="Password"
              type="password"
              name='password'
              autoComplete="current-password"
              margin="normal"
              style={{width:320}}
              onChange={this.handleChange}
            />
          </div>
          <Row style={{marginTop:20}}>
            <Col >
              <span className='text-danger'>{this.state.errorMessage}</span>
              {this.state.loading ? <Button className='float-right' variant="contained" color="secondary" >
                Signing in...
              </Button> :
              <Button className='float-right' type='submit' variant="contained" color="secondary" >
                Sign In
              </Button>}
            </Col>
          </Row>
        </div>
      </form>
    )
  }
}

export default SignIn;
