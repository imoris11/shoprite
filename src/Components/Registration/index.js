import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import PropTypes from 'prop-types';
import Register from './Register';
import SignIn from './SignIn';
function TabContainer(props) {
  return (
    <Typography component="div" style={{ padding: 8 * 3 }}>
      {props.children}
    </Typography>
  );
}

TabContainer.propTypes = {
  children: PropTypes.node.isRequired,
};

const DialogTitle = withStyles(theme => ({
  root: {
    borderBottom: `1px solid ${theme.palette.divider}`,
    margin: 0,
    padding: theme.spacing.unit * 2,
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing.unit,
    top: theme.spacing.unit,
    color: theme.palette.grey[500],
  },
  }))(props => {
  const { children, classes, onClose } = props;
  return (
    <MuiDialogTitle disableTypography className={classes.root}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label="Close" className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
});

const DialogContent = withStyles(theme => ({
  root: {
    margin: 0,
    padding: theme.spacing.unit * 2,
  },
}))(MuiDialogContent);


class CustomizedDialog extends Component {
  state = {
    value:0,
  };
  handleChange = (event, value) => {
    this.setState({ value });
  };
  render() {
    const { value } = this.state;
    return (
      <div>
        <Dialog
          onClose={()=>this.props.handleClose()}
          aria-labelledby="customized-dialog-title"
          open={this.props.showDialog}
        >
          <DialogTitle id="customized-dialog-title" onClose={()=>this.props.handleClose()}>
            <Tabs value={value} onChange={this.handleChange}>
              <Tab label="Sign in" />
              <Tab label="Register" />
            </Tabs>
          </DialogTitle>
          <DialogContent>
            {value === 0 && <TabContainer><SignIn /></TabContainer>}
            {value === 1 && <TabContainer><Register /></TabContainer>}
          </DialogContent>
        </Dialog>
      </div>
    );
  }
}

export default CustomizedDialog;
