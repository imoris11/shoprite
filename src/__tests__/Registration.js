import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import Registration from '../Components/Registration';
import Register from '../Components/Registration/Register';
import SignIn from '../Components/Registration/SignIn';
describe('Test Registration Component', () => {

   it('renders without crashing', () => {
      shallow(<Registration showDialog={true} handleClose={()=>console.log("Here")} />);
    });

    it('renders correctly', ()=> {
      const component = shallow(<Registration showDialog={true} handleClose={()=>console.log("Here")} />);
      expect(component).toMatchSnapshot();
    });

    it('shows register form button', ()=>{
       mount(<Register />);
    });

    it('shows signin form button', ()=>{
       mount(<SignIn />);
    });
});
