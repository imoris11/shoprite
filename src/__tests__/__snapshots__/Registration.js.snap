// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Registration Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomizedDialog
    handleClose={[Function]}
    showDialog={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-labelledby="customized-dialog-title"
        onClose={[Function]}
        open={true}
      >
        <WithStyles(Component)
          id="customized-dialog-title"
          onClose={[Function]}
        >
          <WithStyles(Tabs)
            onChange={[Function]}
            value={0}
          >
            <WithStyles(Tab)
              label="Sign in"
            />
            <WithStyles(Tab)
              label="Register"
            />
          </WithStyles(Tabs)>
        </WithStyles(Component)>
        <WithStyles(WithStyles(DialogContent))>
          <TabContainer>
            <SignIn />
          </TabContainer>
        </WithStyles(WithStyles(DialogContent))>
      </WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "customized-dialog-title",
        "children": Array [
          <WithStyles(Component)
            id="customized-dialog-title"
            onClose={[Function]}
          >
            <WithStyles(Tabs)
              onChange={[Function]}
              value={0}
            >
              <WithStyles(Tab)
                label="Sign in"
              />
              <WithStyles(Tab)
                label="Register"
              />
            </WithStyles(Tabs)>
          </WithStyles(Component)>,
          <WithStyles(WithStyles(DialogContent))>
            <TabContainer>
              <SignIn />
            </TabContainer>
          </WithStyles(WithStyles(DialogContent))>,
        ],
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tabs)
              onChange={[Function]}
              value={0}
            >
              <WithStyles(Tab)
                label="Sign in"
              />
              <WithStyles(Tab)
                label="Register"
              />
            </WithStyles(Tabs)>,
            "id": "customized-dialog-title",
            "onClose": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Tab)
                  label="Sign in"
                />,
                <WithStyles(Tab)
                  label="Register"
                />,
              ],
              "onChange": [Function],
              "value": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Sign in",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Register",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TabContainer>
                <SignIn />
              </TabContainer>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <SignIn />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-labelledby="customized-dialog-title"
          onClose={[Function]}
          open={true}
        >
          <WithStyles(Component)
            id="customized-dialog-title"
            onClose={[Function]}
          >
            <WithStyles(Tabs)
              onChange={[Function]}
              value={0}
            >
              <WithStyles(Tab)
                label="Sign in"
              />
              <WithStyles(Tab)
                label="Register"
              />
            </WithStyles(Tabs)>
          </WithStyles(Component)>
          <WithStyles(WithStyles(DialogContent))>
            <TabContainer>
              <SignIn />
            </TabContainer>
          </WithStyles(WithStyles(DialogContent))>
        </WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "customized-dialog-title",
          "children": Array [
            <WithStyles(Component)
              id="customized-dialog-title"
              onClose={[Function]}
            >
              <WithStyles(Tabs)
                onChange={[Function]}
                value={0}
              >
                <WithStyles(Tab)
                  label="Sign in"
                />
                <WithStyles(Tab)
                  label="Register"
                />
              </WithStyles(Tabs)>
            </WithStyles(Component)>,
            <WithStyles(WithStyles(DialogContent))>
              <TabContainer>
                <SignIn />
              </TabContainer>
            </WithStyles(WithStyles(DialogContent))>,
          ],
          "onClose": [Function],
          "open": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tabs)
                onChange={[Function]}
                value={0}
              >
                <WithStyles(Tab)
                  label="Sign in"
                />
                <WithStyles(Tab)
                  label="Register"
                />
              </WithStyles(Tabs)>,
              "id": "customized-dialog-title",
              "onClose": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Tab)
                    label="Sign in"
                  />,
                  <WithStyles(Tab)
                    label="Register"
                  />,
                ],
                "onChange": [Function],
                "value": 0,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "Sign in",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "Register",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TabContainer>
                  <SignIn />
                </TabContainer>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <SignIn />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              false,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
